dev/io.openliberty.org.apache.myfaces.4.0_fat/test-applications/FacesContextGetLifecycle.war/src/io/openliberty/org/apache/myfaces40/fat/facesContext/getLifecycle/bean/AddPhaseListenerBean.java

```diff
@@ -0,0 +1,35 @@
+/*******************************************************************************
+ * Copyright (c) 2022 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package io.openliberty.org.apache.myfaces40.fat.facesContext.getLifecycle.bean;
+
+import io.openliberty.org.apache.myfaces40.fat.facesContext.getLifecycle.phaseListener.ProgrammaticallyAddedPhaseListener;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.faces.context.FacesContext;
+import jakarta.inject.Named;
+
+/**
+ * This bean is used to add a PhaseListener programmatically using the new FacesContext.getLifecycle() API
+ * in Faces 4.0.
+ */
+@Named
+@ApplicationScoped
+public class AddPhaseListenerBean {
+    boolean phaseListenerAdded = false;
+
+    public void addPhaseListener() {
+        FacesContext facesContext = FacesContext.getCurrentInstance();
+
+        if (!phaseListenerAdded) {
+            ProgrammaticallyAddedPhaseListener phaseListener = new ProgrammaticallyAddedPhaseListener();
+            facesContext.getLifecycle().addPhaseListener(phaseListener);
+        }
```

Comment:
I know this test only runs once, but should phaseListenerAdded be set to true (if testing manually) in the if block?

Reply:
Yes, that was the whole point of doing it this way, I forgot it. Nice catch, I'll fix it.